reputation/microeconomic_reputation.tex

```diff
@@ -0,0 +1,367 @@
+
+In other words, for agent $A$ and decision $D$, $E_A(D)$ is "how happy agent $A$ expects to be if they do $D$".
+\newline
+
+\textbf{Golem} - The observable result of all the work done by the Golem Factory - the Golem protocol, available software, state of the market etc.
+\newline
+
+\subsection{Additional assumptions}
+
+\begin{enumerate}
+\item The Expected Utility Hypothesis\footnote{\href{https://en.wikipedia.org/wiki/Expected\_utility\_hypothesis}{https://en.wikipedia.org/wiki/Expected\_utility\_hypothesis}}: every agent tries to maximize their expected utility.
+
+\item Every agent can costlessy access and analyze all of the publicly available information about Golem. Thanks to this assumption we can remove $I$ from the equations - we no longer
+    care about "what agent knows", but only about "what really is". This is a major simplification that might not be a good approximation of the "real world" Golem market, but we
+    accept it for the sake of the brevity of this document. Also "truth about Golem is known" is an ideal world we would like to live in.
+
+\item The expected utility and money are interchangeable, that is - for every agent $A$ and a pair of decisions $(D_1, D_2): E_A(D_1) < E_A(D_2)$, there is an amount of money X that
+    can be given to the agent so that $E_A(D_1) + U_A(X) = E_A(D_2)$. This way we can treat utilities as if they were money, and thus compare them between different agents.
+
+\item Lets define the Golem Value as:
+
+\begin{equation}
+    V_G = \sum_{A \in all\_agents\_using\_Golem}E_A(USE\_GOLEM)
+\end{equation}
+(Note that - because of assumption 1 - $E_A(USE\_GOLEM)$ is always positive, decision "use Golem" is only made by agents who expect to profit from it).
+
+We assume that maximization of $V_G$ is one of the goals of the Golem Factory - and from the point of view of this document, the only goal.
+\end{enumerate}
+
+\section{The "reputation" and its purpose}
+\subsection{Notation}
+
+We'll be using a bunch of different symbols, but they follow a clear common pattern, so please don't be frightened.
+
+\begin{itemize}
+\item $V_A(a)$ is the total value of the agreement $a$ from the POV of agent $A$. As per assumption 3, this is "utility expressed as money".\footnote{
+I.e. $V_A(a) = X$ means "when agent $A$ takes part in the agreement $a$, their hapiness changes as if they were given $X$ money".}, i.e.:
+    \begin{itemize}
+        \item For the provider it is the utility of the money received decreased by the utility cost of the hardware/electricity/etc.
+        \item For the requestor it is the utility of the resources obtained decreased by the utility cost of the money spent
+    \end{itemize}
+\item $V_P(a)$/$V_R(a)$ are agreement values from the POV of (respectively) provider/requestor.
+\item $V_{PN}(a)$/$V_{RN}(a)$ are nominal (i.e. negotiated) values of the agreement\footnote{
```

Comment:
Here it is even more unclear whether you mean *ex ante* or *ex post* values.If the former, then what is `V_PN` if it stipulates pay per use? If the former, can the agent even estimate `V_PN` when deciding whether to sign the agreement?

Reply:
> Here it is even more unclear whether you mean _ex ante_ or _ex post_ values.

The latter, _ex post_.

> can the agent even estimate `V_PN` when deciding whether to sign the agreement?

I don't see why not? E.g. they can have a history of agreements and estimate `V_PN` as "average V_PN for all agreements similar enough". Or just have some predictions/estimations of the relevant parameters (e.g. agreement length).

Reply:
I've added a general note about all $V*$s being _ex post_ in ac044d4

Reply:
> > can the agent even estimate `V_PN` when deciding whether to sign the agreement?
> 
> I don't see why not? E.g. they can have a history of agreements and estimate `V_PN` as "average V_PN for all agreements similar enough". Or just have some predictions/estimations of the relevant parameters (e.g. agreement length).

It seems substantiating this claim may require a lot of work at the very least (and fail in the end). 

On the other hand your argument here sems to contradict your initial assumption "we can remove $I$ from the equations - we no longer care about what agent knows"

Reply:
> On the other hand your argument here sems to contradict your initial assumption "we can remove $I$ from the equations - we no longer care about what agent knows"

The assumption is about (quote, line 64) **publicly available information** only.
Removing **all** of the agent information from the equations is in obvious contradiction not only with this argument, but also with e.g. _Conclusion 2.3_ that reputation system is based on making new information available.